app/services/account_service.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# Retrieve names from the account API run by UIT
+class AccountService
+  Account = Struct.new(:sunetid, :name, :description, keyword_init: true)
+
+  def self.call(...)
+    new(...).call
+  end
+
+  def initialize(sunetid:)
+    @sunetid = sunetid
+  end
+
+  # @return [Account, nil] the account or nil if not found
+  def call
+    params = Rails.cache.fetch(sunetid, namespace: 'account', expires_in: 1.month) do
```

Comment:
A month seems like a long time to cache, but probably not an issue for sunet -> user...

Reply:
That's what it was in H2.